Source/Dafny/Translator.cs

```diff
@@ -18319,6 +18319,9 @@ private class TriggersExplorer : BottomUpVisitor {
       if (expr is FunctionCallExpr) {
```

Comment:
Does `TypeTestExpr` also need to be checked, since it has a `.ToType` field?

It seems easy to forget to add a check for a type of expression here. If a new expression is added to the language, how would you know about this method?

Do you have an opinion on one of the following alternatives?
- Make `ComputeFreeTypeVariables` a virtual method on `Expression`, this approach makes it easier to discover the method when creating a new type of expression, but doesn't increase the amount of code.
- Add an abstract method `GetSubTypes` to `Expression` which would return `Util.Concat(e.TypeApplication_AtEnclosingClass, e.TypeApplication_JustFunction)` for `FunctionCallExpr`, `e.BoundVars.Select(bv => bv.Type)` for `LetExpr`, and `Enumerable.Empty<Type>()` for all others
- Write a code generator that finds values of type `Microsoft.Dafny.Type` inside each `Expression` type and generates the correct `ComputeFreeTypeVariables` method. With [SourceGenerators](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/) it's easier to run the generator at the right time. We could also use such a generator to generate the `SubExpressions` method and possibly others.

Reply:
Good catch! I added the `.ToType` field.

Yes, this is brittle and can be improved. I prefer the `GetSubTypes` idea above (but avoiding the name `SubType`, which is confusing in this context; how about `GetComponentTypes` or something like that?).